src/node.rs

```diff
@@ -0,0 +1,153 @@
+use crate::helper::Identifier;
+use crate::kbucket::{Bucket, KbucketTable, TableRecord};
+use std::collections::HashMap;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use tokio::io;
+use tokio::net::UdpSocket;
+
+// The main Kademlia client struct.
+// Provides user-level API for performing querie and interacting with the underlying service.
+#[derive(Debug, PartialEq)]
+pub struct Node {
+    pub node_id: Identifier,
+    pub table: KbucketTable,
+    pub store: HashMap<Vec<u8>, Vec<u8>>, // Same storage as Portal network to store samples
+}
+
+impl Node {
+    pub fn new(node_id: Identifier, table_record: TableRecord) -> Self {
```

Comment:
same thing above, you can make a table record if you want but all this fn really needs is the (id, ip, port) data

Reply:
I modified the table record to only contain the ip and port.  But want to leave the TableRecord data structure in-tact in case we'd want to add other data to the record.